src/glibc/lind_syscall/lind_syscall.c

```diff
@@ -1,39 +1,89 @@
-// callname: a legacy argument, will be changed after 3i has integrated
-// arg1-arg6: actual argument of the syscall, note that all the pointers passed here is 32-bit virtual wasm address
-//            and should be handled appropriately. This might be changed later and the address translation might be
-//            handled here instead
-int lind_syscall (unsigned int callnumber, unsigned long long callname, unsigned long long arg1, unsigned long long arg2, unsigned long long arg3, unsigned long long arg4, unsigned long long arg5, unsigned long long arg6, int raw)
+/*
+ * make_threei:
+ *
+ * Unified function used to invoke threei style syscalls.  This is the
+ * core entry point for all syscall transitions into the lind runtime,
+ * including inter-cage calls and grates.  Unlike MAKE_TRADITION, this function
+ * explicitly specifies both `self_cageid` and `target_cageid`, allowing
+ * fine-grained routing of syscalls across cage boundaries.
+ *
+ * Each logical argument is passed in a (value, cageid) pair, enabling
+ * three-i's interposition layer to perform selective rewriting, mediation,
+ * or redirection.  The final argument, `raw_flag`, determines whether
+ * lind_syscall should apply standard POSIX errno translation or return
+ * the raw trampoline result directly.
+ *
+ * make_threei is designed to be the **canonical** macro for all new
+ * inter-cage or grate-level syscall invocations.  Grates and higher-level
+ * components should call make_threei directly rather than relying on
+ * MAKE_TRADITION.
+ *
+ * This function acts as the middle layer between:
+ *   (1) the MAKE_TRADITION macros in glibc, and
+ *   (2) the actual Wasmtime entry function (__lind_make_syscall_trampoline).
+ *
+ * It forwards all syscall parameters—including the inter-cage metadata
+ * (self_cageid, target_cageid, argX_cageid pairs) to the underlying
+ * trampoline, but also optionally performs post-processing on the return
+ * value depending on `raw_flag`.
+ *
+ * The `raw_flag` controls whether this wrapper should apply the standard
+ * errno handling:
+ *
+ *   raw_flag == 0:
+ *       The return value is treated as a complete syscall result.
+ *       Negative values in the range [-255, -1] are interpreted as
+ *       `-errno`, errno is set accordingly, and the wrapper returns -1.
+ *       All other values are returned directly.
+ *
+ *   raw_flag == 1:
+ *       The wrapper does *not* apply any errno translation.
+ *       The raw return value from the trampoline is returned as-is.
+ *
+ * This distinction is required because some syscalls—especially futex-related
+ * operations (e.g., lll_futex_wake, lll_futex_requeue, etc.) expect the
+ * trampoline to return raw -errno value and must not receive additional errno 
+ * post-processing at this layer. Other syscalls, however, rely on the standard 
+ * POSIX errno translation implemented here.
+ */
+int make_threei (unsigned int callnumber, 
```

Comment:
Still nitpicking at names so apologies. Would it make sense to lengthen this to `make_threei_call`

Would make more sense to me since were making a call and not making threei